packages/hardhat-vyper/src/index.ts

```diff
@@ -249,14 +254,26 @@ subtask(TASK_COMPILE_VYPER)
         ({ version }) => version
       );
 
+      const versionsToSettings = Object.fromEntries(
+        config.vyper.compilers.map(({ version, settings }) => [
+          version,
+          settings,
+        ])
+      );
+
       const versionGroups: Record<string, ResolvedFile[]> = {};
       const unmatchedFiles: ResolvedFile[] = [];
 
       for (const file of resolvedFiles) {
+        const settings = versionsToSettings[file.content.versionPragma];
```

Comment:
I'd change this to `const settings = versionsToSettings[file.content.versionPragma] || {};` to avoid this check: `...(settings !== undefined ? { settings } : {})`